File: Python/Mahasiswa.py
# Import file.
from Sivitas_Akademik import Sivitas_Akademik


# membuat kelas Mahasiswa dengan pewarisan dari kelas Sivitas_Akademik
class Mahasiswa (Sivitas_Akademik):
    # Membuat Private attributes.
    __nim = ""
    __fakultas = ""

    # Constructor with default initialization
    def __init__(self, nama="", nik="", Jenis_Kelamin="", asal_univ="", email="", nim="", fakultas=""):
        # memanggil konstruktor dari kelas induk (superclass)
        super().__init__(nik, nama, Jenis_Kelamin, asal_univ, email)
        self.__nim = nim
        self.__fakultas = fakultas

    # Setter and Getter methods for private attributes
    # Set Nim
    def setNim(self, nimMahasiswa):
        self.__nim = nimMahasiswa

    # Get Nim
    def getNim(self):
        return self.__nim

    # Set Fakultas
    def setFakultas(self, fakultas):
        self.__fakultas = fakultas

    # Get Fakultas
    def getFakultas(self):
        return self.__fakultas

File: C++/Mahasiswa.cpp
#include "header.hh"
// Default constructor
Mahasiswa::Mahasiswa() : Sivitas_Akademik()
{
    // initialize private variables
    nim = "";
    fakultas = "";
}

// Overloaded constructor with 5 parameters
Mahasiswa::Mahasiswa(string nama, string nik, string gender, string asal_universitas, string email_edu) : Sivitas_Akademik(nama, nik, gender, asal_universitas, email_edu)
{
}

// Overloaded constructor with 7 parameters
Mahasiswa::Mahasiswa(string nama, string nik, string gender, string asal_universitas, string email_edu, string nim, string fakultas) : Sivitas_Akademik(nama, nik, gender, asal_universitas, email_edu)
{
    // initialize private variables
    this->nim = nim;
    this->fakultas = fakultas;
}

// Getter method for nim
string Mahasiswa::getNim()
{
    return nim;
}

// Setter method for nim
void Mahasiswa::setNim(string nim)
{
    this->nim = nim;
}

// Getter method for fakultas
string Mahasiswa::getFakultas()
{
    return fakultas;
}

// Setter method for fakultas
void Mahasiswa::setFakultas(string fakultas)
{
    this->fakultas = fakultas;
}

// Destructor
Mahasiswa::~Mahasiswa() {}